### PR TITLE
make section related calls iff section exists

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?
More in series and end-slate relies on page section. If the section doesn't exist then the requests 404. See console on https://www.theguardian.com/global/video/2016/nov/01/what-happened-at-the-battle-of-orgreave-video-explainer for example.

This change checks page section exists before making api calls.

## What is the value of this and can you measure success?
Fewer 404s

## Does this affect other platforms - Amp, Apps, etc?
No. Will very likely need to replicate on off platform embed player in a later PR.

## Screenshots
n/a

## Request for comment
@gidsg @markjamesbutler

### CODE REVIEW

File: common/app/model/VideoPlayer.scala [hunk 1]
```diff
@@ -16,6 +16,7 @@ case class VideoPlayer(
   title: String,
   autoPlay: Boolean,
   showControlsAtStart: Boolean,
+  // TODO make Option[String] as relies on section being set
```
Comment (akash1810) [inline common/app/model/VideoPlayer.scala]: if there isn't a section `endSlatePath` will 404 - it shouldn't be on the DOM in this case

**Comment by paperboyo** [comment]
I may be entirely wrong, but maybe it should just call for latest videos from the tag instead of ending up empty? Would be https://www.theguardian.com/politics/miners-strike-1984-85+type/video for the above example.

**Comment by akash1810** [comment]
@paperboyo it doesn't end up empty as the fallback behaviour is to show the most popular videos

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 2]
```diff
@@ -399,7 +403,11 @@ define([
         mediator.on('page:media:moreinloaded', initPlayButtons);
 
         initFacia();
-        initMoreInSection();
+
+        if (config.page.section) {
```
Comment (gidsg) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: There are already logic checks inside `initMoreInSection()` function, so I think this extra check would make more sense to do there?

**Review by gidsg — CHANGES_REQUESTED** [review]

File: static/src/javascripts/bootstraps/enhanced/media/main.js (hunk 2 above)
Comment (akash1810) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: yeah I debated on putting it in there. My argument for putting it here was that the function name `initMoreInSection` sounds like it expects section to exist... it was a weak argument so happy to change.

Comment (akash1810) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: updated.

**Review by gidsg — APPROVED** [review]

**Pull request merged**